monero-oxide/ringct/bulletproofs/src/original/mod.rs

```diff
@@ -278,10 +290,16 @@ impl<'a> AggregateRangeStatement<'a> {
     transcript = Self::transcript_tau_x_mu_t_hat(transcript, proof.tau_x, proof.mu, proof.t_hat);
     let x_ip = transcript;
 
-    proof.A = proof.A.mul_by_cofactor();
-    proof.S = proof.S.mul_by_cofactor();
-    proof.T1 = proof.T1.mul_by_cofactor();
-    proof.T2 = proof.T2.mul_by_cofactor();
+    let decomp_mul_cofactor = |p| decompress_point(p).map(|p| EdwardsPoint::mul_by_cofactor(&p));
+
+    let (Some(A), Some(S), Some(T1), Some(T2)) = (
```

Comment:
Nit: Can perform a destructing assignment off `proof` to ensure we no longer have `proof.A` and `A` in-scope (as `A` will be copied here and could still be used later), by removing `proof` from the scope (as it isn't `Copy`).